src/twisted/conch/ssh/transport.py

```diff
@@ -1275,31 +1279,44 @@ def _ssh_KEX_ECDH_INIT(self, packet):
         pubHostKey = self.factory.publicKeys[self.keyAlg]
         privHostKey = self.factory.privateKeys[self.keyAlg]
 
-        # Get the curve instance
-        try:
-            curve = keys._curveTable[b'ecdsa' + self.kexAlg[4:]]
-        except KeyError:
-            raise UnsupportedAlgorithm('unused-key')
-
-        # Generate the private key
-        ecPriv = ec.generate_private_key(curve, default_backend())
-
-        # Get the public key
-        ecPub = ecPriv.public_key()
-        encPub = ecPub.public_numbers().encode_point()
-
-        # Take the provided public key and transform it into
-        # a format for the cryptography module
-        theirECPub = ec.EllipticCurvePublicNumbers.from_encoded_point(
-                        curve, pktPub).public_key(default_backend())
-
-        # We need to convert to hex,
-        # so we can convert to an int
-        # so we can make it a multiple precision int.
-        sharedSecret = MP(
-                       int(
-                        binascii.hexlify(
-                          ecPriv.exchange(ec.ECDH(), theirECPub)), 16))
+        if self.kexAlg.find(b'25519') != -1:
```

Comment:
Just a nitpick, but, this is perhaps more idiomatically expressed as `if b'25519' in self.kexAlg`